docs/nextcloud.md

```diff
@@ -0,0 +1,425 @@
+It is available with the `cozyMetadata.sourceAccount` of the shortcut file for
+example.
+
+The `*path` parameter is the path of the file/directory on the NextCloud.
+
+**Note:** a permission on `GET io.cozy.files` is required to use this route.
+
+### Request (list)
+
+```http
+GET /remote/nextcloud/4ab2155707bb6613a8b9463daf00381b/Documents HTTP/1.1
+Host: cozy.example.net
+Authorization: Bearer eyJhbG...
+```
+
+### Response (list)
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/vnd.api+json
+```
+
+```json
+{
+  "data": [
+    {
+      "type": "io.cozy.remote.nextcloud.files",
+      "id": "192172",
+      "attributes": {
+        "type": "directory",
+        "name": "Images",
+        "updated_at": "Thu, 02 May 2024 09:29:53 GMT",
+        "etag": "\"66335d11c4b91\""
+      },
+      "meta": {},
+      "links": {
+        "self": "https://nextcloud.example.net/apps/files/files/192172?dir=/Documents"
+      }
+    },
+    {
+      "type": "io.cozy.remote.nextcloud.files",
+      "id": "208937",
+      "attributes": {
+        "type": "file",
+        "name": "BugBounty.pdf",
+        "size": 2947,
+        "mime": "application/pdf",
+        "class": "pdf",
+        "updated_at": "Mon, 14 Jan 2019 08:22:21 GMT",
+        "etag": "\"dd1a602431671325b7c1538f829248d9\""
+      },
+      "meta": {},
+      "links": {
+        "self": "https://nextcloud.example.net/apps/files/files/208937?dir=/Documents"
+      }
+    },
+    {
+      "type": "io.cozy.remote.nextcloud.files",
+      "id": "615827",
+      "attributes": {
+        "type": "directory",
+        "name": "Music",
+        "updated_at": "Thu, 02 May 2024 09:28:37 GMT",
+        "etag": "\"66335cc55204b\""
+      },
+      "meta": {},
+      "links": {
+        "self": "https://nextcloud.example.net/apps/files/files/615827?dir=/Documents"
+      }
+    },
+    {
+      "type": "io.cozy.remote.nextcloud.files",
+      "id": "615828",
+      "attributes": {
+        "type": "directory",
+        "name": "Video",
+        "updated_at": "Thu, 02 May 2024 09:29:53 GMT",
+        "etag": "\"66335d11c2318\""
+      },
+      "meta": {},
+      "links": {
+        "self": "https://nextcloud.example.net/apps/files/files/615828?dir=/Documents"
+      }
+    }
+  ],
+  "meta": {
+    "count": 5
+  }
+}
+```
+
+### Request (download)
+
+```http
+GET /remote/nextcloud/4ab2155707bb6613a8b9463daf00381b/Documents/Wallpaper.jpg?Dl=1 HTTP/1.1
+Host: cozy.example.net
+Authorization: Bearer eyJhbG...
+```
+
+### Response (download)
+
+```http
+HTTP/1.1 200 OK
+Content-Type: image/jpeg
+Content-Length: 12345
+Content-Disposition: attachment; filename="Wallpaper.jpg"
+
+...
+```
+
+#### Status codes
+
+- 200 OK, for a success
+- 401 Unauthorized, when authentication to the NextCloud fails
+- 404 Not Found, when the account is not found or the directory is not found on the NextCloud
+
+## PUT /remote/nextcloud/:account/*path
```

Comment:
This should not be a POST request to add a new file. To rename an existing file, I imagine @nono you were thinking of using the move route ? 🤔